src/Jackalope/Security/AccessControlEntry.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace Jackalope\Security;
+
+use PHPCR\Security\AccessControlEntryInterface;
+use PHPCR\Security\PrincipalInterface;
+use PHPCR\Security\PrivilegeInterface;
+
+/**
+ * {@inheritDoc}
+ */
+class AccessControlEntry implements \IteratorAggregate, AccessControlEntryInterface
+{
+    /**
+     * @var PrincipalInterface
+     */
+    private $principal;
+
+    /**
+     * @var PrivilegeInterface[]
+     */
+    private $privileges;
+
+    public function __construct(PrincipalInterface $principal, array $privileges)
+    {
+        $this->principal = $principal;
+        // TODO: validate privileges
```

Comment:
what i meant is validate that its actually PrivilegeInterface instances, in case somebody tries to set them with strings or something like that.